src/main/java/htsjdk/samtools/SAMFileHeader.java

```diff
@@ -150,6 +151,7 @@ public SAMSequenceDictionary getSequenceDictionary() {
      * Look up sequence record by name.
      */
     public SAMSequenceRecord getSequence(final String name) {
+        if (mSequenceDictionary == null) throw new TribbleException.InvalidHeader("sequence dictionary is empty");
```

Comment:
Having looked at this a bit more, I don't think we want to throw here. As far as I can tell its not required that you have a sequence dictionary, and the current implementation allows you to set it to null on the header, so throwing InvalidHeader is probably too strong (and we generally don't use the Tribble exceptions from within the samtools package anyway). Since it looks like there are already cases where this method is called and the return value is checked for null, I think we should just make this method return null if the sequence dictionary is null, and update the javadoc to make that explicit. Then IntervalList will throw an IllegalArgumentException instead of getting an NPE.

Also, for new code we generally like to put the if statement body on a separate line, with braces, even if its a single statement. To just return null, a ternary might be even better.